servers/src/test/java/tachyon/worker/block/TieredBlockStoreTestUtils.java

```diff
@@ -77,40 +91,120 @@ public static TachyonConf newTachyonConf(int[] tierLevel, StorageLevelAlias[] ti
     Preconditions.checkArgument(tierLevel.length == tierCapacity.length,
         "tierCapacity and tierLevel should have the same length");
     int nTier = tierLevel.length;
-    for (int i = 0; i < nTier; i ++) {
-      Preconditions.checkArgument(tierPath[i].length == tierCapacity[i].length,
-          String.format("tierPath[%d] and tierCapacity[%d] should have the same length", i, i));
+
+    tierPath = concatAndCreateDir(baseDir, tierPath);
+    if (workerDataFolder != null) {
+      sTachyonConf.set(Constants.WORKER_DATA_FOLDER, workerDataFolder);
     }
+    sTachyonConf.set(Constants.WORKER_MAX_TIERED_STORAGE_LEVEL, String.valueOf(nTier));
 
-    TachyonConf tachyonConf = new TachyonConf();
-    tachyonConf.set(Constants.WORKER_MAX_TIERED_STORAGE_LEVEL, String.valueOf(nTier));
+    // set up each tier in turn
     for (int i = 0; i < nTier; i ++) {
-      int level = tierLevel[i];
-      tachyonConf.set(String.format(Constants.WORKER_TIERED_STORAGE_LEVEL_ALIAS_FORMAT, level),
-          tierAlias[i].toString());
-
-      StringBuilder sb = new StringBuilder();
-      for (String path : tierPath[i]) {
-        sb.append(path);
-        sb.append(",");
-      }
-      tachyonConf.set(String.format(Constants.WORKER_TIERED_STORAGE_LEVEL_DIRS_PATH_FORMAT, level),
-          sb.toString());
+      setTachyonConfOneTier(tierLevel[i], tierAlias[i], tierPath[i], tierCapacity[i]);
+    }
+    return sTachyonConf;
+  }
 
-      sb = new StringBuilder();
-      for (long capacity : tierCapacity[i]) {
-        sb.append(capacity);
-        sb.append(",");
-      }
-      tachyonConf.set(
-          String.format(Constants.WORKER_TIERED_STORAGE_LEVEL_DIRS_QUOTA_FORMAT, level),
-          sb.toString());
+  /**
+   * Setup a {@link TachyonConf} for a {@link TieredBlockStore} with only *one tier* configured
+   * by the parameters. For simplicity, you can use {@link #setTachyonConfDefault(String)} which
+   * sets up the tierBlockStore with default values.
+   *
+   * @param baseDir the directory path as prefix for all the paths of directories in the tiered
+   *        storage. When specified, the directory needs to exist before calling this method.
+   * @param tierLevel level of this tier.
+   * @param tierAlias alias of this tier.
+   * @param tierPath path of this tier. When `baseDir` is specified, the actual test tierPath
+   *        turns into `baseDir/tierPath`.
+   * @param tierCapacity capacity of this tier.
+   * @param workerDataFolder when specified it sets up the tachyon.worker.data.folder property.
+   * @return the created TachyonConf.
+   * @throws Exception when error happens during creating temporary folder.
+   */
+  public static TachyonConf setTachyonConf(String baseDir, int tierLevel, StorageLevelAlias
```

Comment:
A general rule for overloading: we shall only use overloading methods if two methods are doing exactly the same work except they take different parameters. This rule doesn't apply here as this `setTachyonConf` has different assumptions as the previous one.

how about renaming this to `setTachyonConfWithSingleTier` and above one as `setTachyonConfWithMultiTier`